README.md

```diff
@@ -194,6 +194,18 @@ parse('<p><br id="remove"></p>', {
 });
 ```
 
+### library
+
+The `library` option allows you to specify which component library is used to create elements (currently either React or Preact). React is used by default if this option is not specified, or if the given option is unknown.
+
+Here's an example showing how to use Preact:
+
+```js
+parse('<br>', {
+  library: 'preact'
```

Comment:
I like your proposition @davidbailey00

What do you think of passing the library module as an option? For example:

```js
// html to react
parse('<p>', {
  library: require('react') // default
});

// html to preact
parse('<p>', {
  library: require('preact')
});

// html to custom
parse('<p>', {
  library: {
    cloneElement: () => { /* ... */ },
    createElement: () => { /* ... */ },
    isValidElement: () => { /* ... */ }
  }
});
```

This way, the API is flexible in terms of [SOLID](https://en.wikipedia.org/wiki/SOLID) principles and it's easier to support new libraries in the future.

Also, do you prefer naming the new option `library` or `react`?